server/src/main/java/io/crate/planner/statement/CopyFromPlan.java

```diff
@@ -118,9 +124,15 @@ public void executeOrFail(DependencyCarrier dependencies,
         NodeOperationTree nodeOpTree = NodeOperationTreeGenerator
             .fromPlan(plan, dependencies.localNodeId());
 
-        dependencies.phasesTaskFactory()
-            .create(plannerContext.jobId(), List.of(nodeOpTree))
-            .execute(consumer, plannerContext.transactionContext());
+        JobLauncher jobLauncher = dependencies.phasesTaskFactory()
+            .create(plannerContext.jobId(), List.of(nodeOpTree));
+
+        if (waitForCompletion) {
+            jobLauncher.execute(consumer, plannerContext.transactionContext());
+        } else {
+            jobLauncher.execute(new EmptyRowConsumer(), plannerContext.transactionContext());
+            consumer.accept(InMemoryBatchIterator.of(Row1.ROW_COUNT_UNKNOWN, SENTINEL), null);
+        }
```

Comment:
To make this testable, you could create a method with a `BiFunction` parameter to pass in the jobLauncher and in the test case a testing one to avoid creating the `DependencyCarrier` or `JobLauncher` in a unit test.
Then the unit test could verify that the given consumer is called with `-1`.

Reply:
Sorry didn't understand it. Could you rephrase?

Reply:
Made some changes according to your review.